pkg/controller/supportbundlecollection/controller.go

```diff
@@ -0,0 +1,986 @@
+		supportBundleCollectionStore:        supportBundleCollectionStore,
+		supportBundleCollectionAppliedToStore: cache.NewIndexer(getSupportBundleCollectionKey, cache.Indexers{
+			nodeSelectorIndex:                nodeSelectorIndexFunc,
+			allNodesIndex:                    allNodesIndexFunc,
+			nodeNameIndex:                    nodeNameIndexFunc,
+			externalNodeSelectorIndex:        externalNodeSelectorIndexFunc,
+			allExternalNodesInNamespaceIndex: allExternalNodesInNamespaceIndexFunc,
+			externalNodeNameIndex:            externalNodeNameIndexFunc,
+			processingNodesIndex:             processingNodesIndexFunc,
+			processingExternalNodesIndex:     processingExternalNodesIndexFunc,
+		}),
+	}
+	c.supportBundleCollectionInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addSupportBundleCollection,
+			UpdateFunc: c.updateSupportBundleCollection,
+			DeleteFunc: c.deleteSupportBundleCollection,
+		},
+		resyncPeriod)
+	nodeInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addNode,
+			DeleteFunc: c.deleteNode,
+		},
+		resyncPeriod)
+	externalNodeInformer.Informer().AddEventHandlerWithResyncPeriod(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.addExternalNode,
+			DeleteFunc: c.deleteExternalNode,
+		},
+		resyncPeriod)
+	return c
+}
+
+// Run will create defaultWorkers workers (goroutines) which will process the SupportBundle events from the work queue.
+func (c *Controller) Run(stopCh <-chan struct{}) {
+	defer c.queue.ShutDown()
+
+	klog.InfoS("Starting", "controllerName", controllerName)
+	defer klog.InfoS("Shutting down", "controllerName", controllerName)
+
+	if !cache.WaitForNamedCacheSync(controllerName, stopCh, c.supportBundleCollectionListerSynced, c.nodeListerSynced, c.externalNodeListerSynced) {
+		return
+	}
+	if err := c.reconcileSupportBundleCollections(); err != nil {
+		klog.ErrorS(err, "Failed to reconcile SupportBundleCollection")
+		return
+	}
+
+	for i := 0; i < defaultWorkers; i++ {
+		go wait.Until(c.worker, time.Second, stopCh)
+	}
+
+	<-stopCh
+}
+
+func (c *Controller) enqueueSupportBundleCollection(obj interface{}, logPrefix string) {
+	bundleCollection := obj.(*v1alpha1.SupportBundleCollection)
+	c.queue.Add(bundleCollection.Name)
+	klog.InfoS(logPrefix, "name", bundleCollection.Name)
+}
+
+func (c *Controller) addSupportBundleCollection(obj interface{}) {
+	bundleCollection := obj.(*v1alpha1.SupportBundleCollection)
+	if isCollectionCompleted(bundleCollection) {
+		klog.InfoS("Processed SupportBundleCollection Add Event, collection is completed")
+		return
+	}
+	c.enqueueSupportBundleCollection(obj, "Enqueued SupportBundleCollection ADD event")
+}
+
+// updateSupportBundleCollection adds the SupportBundleCollection name into queue if the conditions are updated. The
+// changes in SupportBundleCollection.Spec is ignored.
```

Comment:
How about adding one more sentence ", as we do not support Spec changes after the collection is started".

Reply:
added